xdg-ninja.sh

```diff
@@ -195,23 +271,26 @@ check_file() {
 
 # Reads files from programs/, calls check_file on each file specified for each program
 do_check_programs() {
+    [ "$OUTPUT_STYLE" = "json" ] && mkdir "$XDG_RUNTIME_DIR"/xdg-ninja/
```

Comment:
Logic of outputting json should be decoupled

Reply:
> While the output format is totally different most of the logic is still the exact same.

Then why are `[ "$OUTPUT_STYLE" = "json" ]` tests are everywhere?

Reply:
There are 5, this is FAR less then the amount of code that would be added by splitting each into their own functions.

Reply:
It is a lot. Those make code messy and unreadable